ts/screens/wallet/AddCardScreen.tsx

```diff
@@ -0,0 +1,222 @@
+/**
+ * Screen for entering the credit card details
+ * (holder, pan, cvc, expiration date)
+ */
+import {
+  Body,
+  Button,
+  Container,
+  Content,
+  Item,
+  Text,
+  View
+} from "native-base";
+import * as React from "react";
+import { FlatList, Image, ScrollView, StyleSheet } from "react-native";
+import { NavigationScreenProp, NavigationState } from "react-navigation";
+
+import I18n from "../../i18n";
+import ROUTES from "../../navigation/routes";
+
+import { none, Option, some } from "fp-ts/lib/Option";
+import { Col, Grid } from "react-native-easy-grid";
+import { cardIcons } from "../../components/wallet/card/Logo";
+
+import _ from "lodash";
+import { Left } from "native-base";
+import { LabelledItem } from "../../components/LabelledItem";
+import { WalletStyles } from "../../components/styles/wallet";
+import AppHeader from "../../components/ui/AppHeader";
+import Icon from "../../theme/font-icons/io-icon-font";
+import variables from "../../theme/variables";
+import { fixExpirationDate, fixPan } from "../../utils/input";
+
+type Props = Readonly<{
+  navigation: NavigationScreenProp<NavigationState>;
+}>;
+
+type State = Readonly<{
+  pan: Option<string>;
+  expirationDate: Option<string>;
+  securityCode: Option<string>;
+  holder: Option<string>;
+}>;
+
+const styles = StyleSheet.create({
+  noBottomLine: {
+    borderBottomWidth: 0
+  },
+  addCardImage: {
+    width: 60,
+    height: 45,
+    resizeMode: "contain",
+    marginTop: 5
+  },
+  verticalSpacing: {
+    width: 16,
+    flex: 0
+  }
+});
+
+const CARD_LOGOS_COLUMNS = 4;
+
+export class AddCardScreen extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
```

Comment:
Remember we are using redux form

Reply:
Gotcha, I will be using it when working on the (upcoming) "card adding" process :)